### DISCUSSION TITLE
Request deduplication

### DISCUSSION THREAD

**Discussion by AjaxSolutions** [pmndrs/jotai#1160] [Q&A] [answered]
Request deduplication is an advanced feature in React Query or SWR.

I assume this is something that Jotai should be able easily handle because the useAtom hook would be using the same async data fetching atom...?

https://github.com/tannerlinsley/react-query/discussions/608
https://swr.vercel.app/docs/advanced/performance#deduplication

**Comment by dai-shi** [comment, marked as answer]
I don't know. It's really how react expects (`read` should be pure). One solution is `jotai/query`.
Wait, what do you mean by "Request deduplication"? Looking at two links, it's something jotai already handles.

**Comment by AjaxSolutions** [comment]
Thanks, this is what I thought.  This is no big deal for Jotai to handle.
